src/arrayMethodJoin.js

```diff
@@ -4,8 +4,19 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultStr = '';
+    const separatorItem = '' + separator;
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === undefined || this[i] === null) {
+        resultStr += separatorItem;
+      } else {
+        resultStr += this[i] + separatorItem;
+      }
+    }
+
+    return resultStr.slice(0, resultStr.length - separatorItem.length);
```

Comment:
Will be better to add some check in the cycle: if it's not the last element - add separator
If it's last element - don't add separator